packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -817,38 +1162,83 @@ impl<'m> RunnerImpl<'m> {
                 };
                 builder = builder.add_buffer(offset_buffer);
 
-                let child_data: mv8::Array<'_> = obj.get("child_data")?;
+                let child_data = new_js_string(scope, "child_data")?;
+
+                let child_data: v8::Local<'s, v8::Array> = obj
+                    .get(scope, child_data.into())
+                    .ok_or_else(|| {
+                        JSRunnerError::V8("Could not get child_data property on obj".to_string())
+                    })?
+                    .try_into()
+                    .map_err(|err| {
+                        JSRunnerError::V8(format!(
+                            "Could not convert child_data from Value to Array: {err}"
+                        ))
+                    })?;
+
+                let child = child_data.get_index(scope, 0).ok_or_else(|| {
+                    JSRunnerError::V8("Could not access index 0 on child_data".to_string())
+                })?;
```

Comment:
Getting the child data should be moved into a function to avoid code duplication here.

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/02af15786b564ada4f9992194281ff0a3281afb9